pkg/kubectl/resource_printer.go

```diff
@@ -442,33 +458,40 @@ func printPod(pod *api.Pod, w io.Writer) error {
 	return nil
 }
 
-func printPodList(podList *api.PodList, w io.Writer) error {
+func printPodList(podList *api.PodList, fn func([]string) error, withNamespace bool) error {
 	for _, pod := range podList.Items {
-		if err := printPod(&pod, w); err != nil {
+		if err := printPod(&pod, fn, withNamespace); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func printPodTemplate(pod *api.PodTemplate, w io.Writer) error {
+func printPodTemplate(pod *api.PodTemplate, fn func([]string) error, withNamespace bool) error {
+	var name string
+	if withNamespace {
+		name = types.NamespacedName{pod.Namespace, pod.Name}.String()
+	} else {
+		name = pod.Name
+	}
+
 	containers := pod.Template.Spec.Containers
 	var firstContainer api.Container
 	if len(containers) > 0 {
 		firstContainer, containers = containers[0], containers[1:]
 	}
-	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\n",
-		pod.Name,
+	err := fn([]string{
```

Comment:
In general code review should catch that and it's usually obvious in output.  I'd say it adds enough complexity that it's probably easier to read removed.

----- Original Message -----
> >  	containers := pod.Template.Spec.Containers
> >  	var firstContainer api.Container
> >  	if len(containers) > 0 {
> >  		firstContainer, containers = containers[0], containers[1:]
> >  	}
> > -	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\n",
> > -		pod.Name,
> > +	err := fn([]string{
> 
> Actually not required. I think it's easy to mistake for specifying format
> (e.g. the number of tab), so just keep it. Though it's not strong opinion, I
> can revert.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7024/files#r30734630